uberfire-services/uberfire-services-api/src/main/java/org/guvnor/common/services/shared/test/Failure.java

```diff
@@ -3,24 +3,26 @@
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
- * 
+ *
  *      http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
-*/
+ */
 
 package org.guvnor.common.services.shared.test;
 
 import org.jboss.errai.common.client.api.annotations.Portable;
+import org.uberfire.backend.vfs.Path;
 
 @Portable
 public class Failure {
 
     private String message;
+    private Path path;
```

Comment:
@gitgabrio Heh, well it will be something that changes for sure in Submarine!

Everywhere and anywhere where `Path` is used will need to change.. it might be that the integration of Submarine and the different containers use a new abstract concept of `Path` but it'd be quite different to the `Path` class we use today (it'd be more of a URI I guess)... but, as with most things "Submarine" I am guessing as to what may be done - given our growing list of requirements from the _containers_!

Reply:
If we change this to String then it is a more error prone identifier that then needs to be changed back to Path. Also when we move away from Path we might miss this since it is not a Path. 

Why Path is used here is because it is an unique identifier for the file. If we move away from using Path we need another identifier and I think String is too weak to be one.

Reply:
@Rikkola @gitgabrio Yes, I'd keep as `Path` in `master`... but when this gets ported to Submarine there will have to be changes in how it operates... Submarine has no concept of `Path` only _content_. However we've raised the requirement to the foundation team that we're going to need some way to get information about other files in the _container_ somehow... no answer has been given yet.

Reply:
@Rikkola @manstis 
I've raised the topic on this PR because I think it will be more and more urgent to change the way we approach the code while moving towards Submarine. It seems that everything worked until now only because from a _functional_ point of view BC is a huge _monolith_, so it does not really matter what dependency yuo put and where - BC has everything inside. But now we are going to need a more clean approach, so we have to think differently, to enforce decoupling between repos - maybe adding new abstractions, or whatever.
Anyway, of course it is fine for me to merge this as currently is.